Feature/dataset model

anusharanganathan opened the pull request:
dataset model similar to dataset model in LoadStone. The spreadsheet DatasetFieldsComparison.ods in Google drive has the comparison of default metadata fields in Sufia and the requirements from loadstone.

I am yet to do adminMetadata and Funder (still searching for appropriate vocabulary).

I have gone for a very simple model for rdf here and may not be true to the requirements laid down by the vocabulary for its usage (especially person). We can enhance this when we have requirements. I think for prrof of concept, this is fine. Happy to discuss this though and make changes as needed.

martyn-w commented:
Hi @anusharanganathan,

There is a bit of repetition in some of the models which could be simplified by including a module which defines the methods below:

* `def final_parent`
* `def persisted?`
* `def new_record?`

anusharanganathan commented:
Hi @martyn-w, I've moved the common methods to a module and inherited them in all of the nested metadata classes. 

I also took the opportunity and removed tests for the nested attributes validation and deletion from the rspec for the nested metadata model concerns. 

These tests appear with the dataset model in which the nested associations are defined along with the conditions for validation and deletion.

martyn-w commented:
Tested and working 👍
